### ISSUE TITLE
Error on remote site (not local) when dealing with a new SQL table

### ISSUE THREAD

**Issue by lquantz** [volkanceylan/Serenity#1798]
Has anyone seen this before?  I added a new SQL table to my database and used Sergen to generate the necessary classes/repos/forms etc for it to be added in my app.

This works fine, at least locally.  I can use my Serenity portal to browse and add new records to that table just fine...as long as I'm doing this from localhost.

If I do the same from a copy of the site running in the cloud (Azure, in my instance) I get an error message in serenity stating 'Invalid object name dbo.MyNewTable' whenever I try touching this new SQL table.

I've pointed my local version of the website to the same Azure SQL instance I'm running the cloud site from, and don't get any errors locally.  Only the cloud website has the problem, and I published my website to Azure so that both the cloud and my local box have the same codebase running.

  I stopped and restarted the website on the Azure portal, to see if there was a caching issue.  But the error remains.  Does anyone have a guess what might be going on?

**Comment by stixoffire** [comment]
@lquantz permissions ? also note dbo...  your azure website has a different identity probably also do you have users in the Azure website and have set in Administration an authorization for that table ?
But permissions is what I would like at as the object is valid - except if you have been denied access in that case it is invalid.

**Comment by lquantz** [comment]
Thans @stixoffire, let me double-check the permissions for that table vs the others.  Nothing comes to mind as to why it would be different, but since it *is* a new table that's a good area to investigate.

**Comment by volkanceylan** [comment]
Check migrations are running.

**Comment by lquantz** [comment]
Migrations are running, I commented out the code that might prevent them from doing so and re-published the site.  

Same problem as before, works fine locally but on Azure I get that message.  Even when pointing to the same database instances locally and on Azure.

**Comment by volkanceylan** [comment]
Migrations are running but you don't have the table, doesn't seem logical to me, as long as you didn't version some migrations with same Version

**Comment by lquantz** [comment]
I'm not sure why migrations would affect my tables, as the existing ones are for the users tables and Northwind.  When I first set things up my tables already existed in SQL Server, I ran sergen against them to generate the row/column/form classes, and was on my way.  No migrations code needed.

With this new table I added it to SQL server, ran sergen against it as before, and got the row/column/form classes again, with no new migrations code added.  

So I'm not sure how migrations would enter into this, unless Serenity relies on it for all tables for a reason I'm not following?

**Comment by volkanceylan** [comment]
Nope Serenity doesnt rely on migrations. If so you are not hitting same db.

**Comment by lquantz** [comment]
You're correct, and I just found out the problem.   The bosses wanted to change the database name to something easier to remember (we had been using an Azure-generated name, like "mydatabase83jsuue") so we made a copy of that db with the new name, and updated the web.config appropriately.

_However_, being geniuses we neglected to change the VS publish options, so we were still pointing to the old database in Azure when publishing, having forgotten that web.config changes themselves aren't enough in that circumstance.  Now everything works fine, and as Homer Simpson would say:

![D'oh](http://i.giphy.com/TwtXMS5EnKDBK.gif)
